esphome/components/template/select/__init__.py

```diff
@@ -0,0 +1,70 @@
+from esphome import automation
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import select
+from esphome.const import (
+    CONF_ID,
+    CONF_INITIAL_VALUE,
+    CONF_LAMBDA,
+    CONF_OPTIONS,
+    CONF_OPTIMISTIC,
+    CONF_RESTORE_VALUE,
+)
+from .. import template_ns
+
+TemplateSelect = template_ns.class_(
+    "TemplateSelect", select.Select, cg.PollingComponent
+)
+
+CONF_SET_ACTION = "set_action"
+
+
+def validate_initial_value_in_options(config):
+    if CONF_INITIAL_VALUE in config:
+        if config[CONF_INITIAL_VALUE] not in config[CONF_OPTIONS]:
+            raise cv.Invalid(
+                f"initial_value '{config[CONF_INITIAL_VALUE]}' is not a valid option [{', '.join(config[CONF_OPTIONS])}]"
+            )
```

Comment:
```suggestion
            )
    return config
```
To fix tests CI error